Keep Patient / Study data of the input.

sogorla1 opened the pull request:
By default, keep the Patient / Study related DICOM metadata from the input images.

JoHof reviewed (changes requested):
Thanks a lot! This is some high quality code and a very useful addition to the package. I haven't put enough attention on the DICOM output. This is much better now. I have added two small comments. Can you please adapt the unit test for the `load_input_image` and `read_dicoms` function this the follwing please? 

```
@pytest.mark.parametrize("read_metadata,exp_len_metadata", [(True, 22),(False, 0)])
def test_read_dicoms(read_metadata, exp_len_metadata):
    d = read_dicoms(os.path.join(os.path.dirname(__file__), "testdata"), read_metadata=read_metadata)
    assert d[0].GetSize() == (512, 512, 2)
    assert len(d[0].GetMetaDataKeys()) == exp_len_metadata
 ```

sogorla1 commented:
Hi Johannes,

Thanks for the compliment, this is my very first python code more than a simple tool 😊

Added the two modifications you asked for and also adjusted the unit tests, the output shows both cases:

pytest -v test_utils.py

============================= test session starts ==============================

platform darwin -- Python 3.9.6, pytest-8.1.1, pluggy-1.4.0 -- /Library/Developer/CommandLineTools/usr/bin/python3

cachedir: .pytest_cache

rootdir: /Users/sogorla/Documents/PDx/lungmask

configfile: pyproject.toml

collected 10 items



test_utils.py::test_bbox_3D PASSED                                       [ 10%]

test_utils.py::test_read_dicoms[True-22] PASSED                          [ 20%]

test_utils.py::test_read_dicoms[False-0] PASSED                          [ 30%]

test_utils.py::test_simple_bodymask PASSED                               [ 40%]

test_utils.py::test_crop_and_resize PASSED                               [ 50%]

test_utils.py::test_preprocess PASSED                                    [ 60%]

test_utils.py::test_reshape_mask PASSED                                  [ 70%]

test_utils.py::test_load_input_image[True-22] PASSED                     [ 80%]

test_utils.py::test_load_input_image[False-0] PASSED                     [ 90%]

test_utils.py::test_postprocessing PASSED                                [100%]



============================== 10 passed in 2.34s ==============================

Now I know how to write python unit tests 😊

I also corrected a small bug to set the keeppatinfo variable, the correct is “keeppatinfo = not args.removemetadata and not args.noHU”

Thanks,

Laszlo



From: Johannes Hofmanninger ***@***.***>
Reply to: JoHof/lungmask ***@***.***>
Date: Saturday, 23 March 2024 at 09:01
To: JoHof/lungmask ***@***.***>
Cc: "Sogor, Laszlo" ***@***.***>, Author ***@***.***>
Subject: Re: [JoHof/lungmask] Keep Patient / Study data of the input. (PR #104)

WARNING: This email originated from outside of GE HealthCare. Please validate the sender's email address before clicking on links or attachments as they may not be safe.

@JoHof requested changes on this pull request.

Thanks a lot! This is some high quality code and a very useful addition to the package. I haven't put enough attention on the DICOM output. This is much better now. I have added two small comments. Can you please adapt the unit test for the load_input_image and read_dicoms function this the follwing please?

@pytest.mark.parametrize("read_metadata,exp_len_metadata", [(True, 22),(False, 0)])

def test_read_dicoms(read_metadata, exp_len_metadata):

    d = read_dicoms(os.path.join(os.path.dirname(__file__), "testdata"), read_metadata=read_metadata)

    assert d[0].GetSize() == (512, 512, 2)

    assert len(d[0].GetMetaDataKeys()) == exp_len_metadata

________________________________

In lungmask/__main__.py<https://github.com/JoHof/lungmask/pull/104#discussion_r1536590073>:

> @@ -81,6 +81,11 @@ def main():

         action="version",

         version=version,

     )

+    parser.add_argument(

+        "--nokeeppatinfo",

please let us call it --removemetadata

________________________________

In lungmask/utils.py<https://github.com/JoHof/lungmask/pull/104#discussion_r1536590366>:

> @@ -15,6 +15,21 @@

 from lungmask.logger import logger





+dicom_metadata_to_keep = (

all upper case please

—
Reply to this email directly, view it on GitHub<https://github.com/JoHof/lungmask/pull/104#pullrequestreview-1956338226>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BHEMIP6DHL33FLPAFVYB3K3YZUZFLAVCNFSM6AAAAABE7JFYLWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSNJWGMZTQMRSGY>.
You are receiving this because you authored the thread.Message ID: ***@***.***>

sogorla1 commented on lungmask/__main__.py:
```diff
@@ -81,6 +81,11 @@ def main():
         version=version,
     )
+    parser.add_argument(
+        "--nokeeppatinfo",
```
Done.

sogorla1 commented on lungmask/utils.py:
```diff
@@ -15,6 +15,21 @@
 from lungmask.logger import logger
 
 
+dicom_metadata_to_keep = (
```
Done.

sogorla1 commented:
Implemented the unit tests as well.

JoHof reviewed (approved):
Approved! thanks a lot.
